core/common/src/serializers/TimeZoneSerializers.kt

```diff
@@ -23,21 +24,35 @@ public object TimeZoneSerializer: KSerializer<TimeZone> {
 
 }
 
-public object ZoneOffsetSerializer: KSerializer<ZoneOffset> {
+public object FixedOffsetTimeZoneSerializer: KSerializer<FixedOffsetTimeZone> {
 
-    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("ZoneOffset", PrimitiveKind.STRING)
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("FixedOffsetTimeZone", PrimitiveKind.STRING)
 
-    override fun deserialize(decoder: Decoder): ZoneOffset {
+    override fun deserialize(decoder: Decoder): FixedOffsetTimeZone {
         val zone = TimeZone.of(decoder.decodeString())
-        if (zone is ZoneOffset) {
+        if (zone is FixedOffsetTimeZone) {
             return zone
         } else {
             throw SerializationException("Timezone identifier '$zone' does not correspond to a fixed-offset timezone")
         }
     }
 
-    override fun serialize(encoder: Encoder, value: ZoneOffset) {
+    override fun serialize(encoder: Encoder, value: FixedOffsetTimeZone) {
         encoder.encodeString(value.id)
     }
 
 }
+
+public object UtcOffsetSerializer: KSerializer<UtcOffset> {
```

Comment:
We could also provide an alternative serializer that stores offset as an integer primitive or a single component object.